game-core/src/main/java/games/strategy/engine/data/Unit.java

```diff
@@ -96,18 +96,10 @@
   private boolean chargedFlatFuelCost = false;
 
   /** Creates new Unit. Owner can be null. */
-  public Unit(final UnitType type, final GamePlayer owner, final GameData data) {
-    this(type, owner, data, UUID.randomUUID());
-  }
-
-  public Unit(final UnitType type, final GamePlayer owner, final GameData data, final UUID id) {
+  public Unit(final UnitType type, @Nullable final GamePlayer owner, final GameData data) {
     super(data);
-
-    checkNotNull(type);
-    checkNotNull(id);
-
-    this.type = type;
-    this.id = id;
+    this.type = checkNotNull(type);
+    this.id = UUID.randomUUID();
 
     setOwner(owner);
```

Comment:
side-comment: @trevan I somewhat wonder if `originalowner` should be set in the constructor so that it is always non-null. It does look like `owner` should be non-null as it defaults to `null_player_id`. Hence, it's unclear if serialization has something to do with the nullity of unit owners.

Reply:
`originalOwner` looks to be used in extremely few places (basically, ownership of factories) so making it non-null is probably a good idea.  I would assume it should default to the value of `owner`.  But I don't see how that affects how `owner` becomes null.

Reply:
>  But I don't see how that affects how owner becomes null.

It doesn't, at least not directly that I know of. If there is somewhere that uses original owner in place of owner, then perhaps..